src/components/VideoPlayer/BaseVideoPlayer.tsx

```diff
@@ -172,12 +176,23 @@ function BaseVideoPlayer({
         });
     }, [currentVideoPlayerRef, handleFullscreenUpdate, handlePlaybackStatusUpdate]);
 
+    useLayoutEffect(() => {
+        return () => {
+            if(!shouldUseSharedVideoElement && (isUploading || videoPlayerRef.current === currentVideoPlayerRef.current)) {
+                currentVideoPlayerRef.current = null;
+            }
+        }
+    }, []);
+
     useEffect(() => {
         if (!isUploading || !videoPlayerRef.current) {
             return;
         }
 
         // If we are uploading a new video, we want to immediately set the video player ref.
+        if(currentVideoPlayerRef.current) {
+            pauseVideo();
+        }
```

Comment:
Why is that needed?

Reply:
Currently, previous playing video is paused when playing new video.
When playing new video, below two functions are called in order and playing video is paused.

- pause video
https://github.com/Expensify/App/blob/c7d692662d069ef8d9c756833767c47e1f7410a2/src/components/VideoPlayerContexts/PlaybackContext.tsx#L41-L49
- update current ref
https://github.com/Expensify/App/blob/c7d692662d069ef8d9c756833767c47e1f7410a2/src/components/VideoPlayerContexts/PlaybackContext.tsx#L51-L62

But when uploading, first function is called after current ref is updated because we set it below.
https://github.com/Expensify/App/blob/c7d692662d069ef8d9c756833767c47e1f7410a2/src/components/VideoPlayer/BaseVideoPlayer.tsx#L180-L181
And previous video is not paused.  So I paused it before setting current ref.